witness/src/Dscp/DB/CanProvideDB/Rocks.hs

```diff
@@ -0,0 +1,41 @@
+
+module Dscp.DB.CanProvideDB.Rocks (RocksDB) where
+
+import Codec.Serialise
+
+import Loot.Base.HasLens (HasLens(lensOf))
+
+import Dscp.DB.CanProvideDB.Class  as Abstract
+import Dscp.DB.CanProvideDB.Common
+import Dscp.DB.Rocks as Rocks
+import Dscp.Util.Serialise (serialise')
+
+instance HasRocksDB ctx m => ProvidesPlugin m where
```

Comment:
Having two completely overlapping instances for the same typeclass looks mad when we could instead carry your `Plugin` in `HasLens` as it's done everywhere else, I suggest you made this on purpose?

Reply:
I've actually removed both instances in the last patch.

Reply:
Now there is just a `Plugin` in both of the contexts.

Reply:
:dancer:

Reply:
Is it possible to merge `WitnessWorkMode` and `WitnessTestMode` back now?